server/core/src/irodsReServer.cpp

```diff
@@ -77,6 +78,30 @@ namespace {
         }
     }
 
+    void load_available_executors(){
+        auto exec = irods::get_server_property<std::unordered_map<std::string, int> >(irods::HOSTS_AVAILABLE_RULE_EXECUTORS);
+        for(const auto &[hostname, port] : exec){
```

Comment:
i think we know the port... they're in the same zone as us... so they have to be using the same port...

hmm, pretty sure about that... @korydraughn @alanking 

if so, don't need a `std::pair`... just the hostnames.

Reply:
i'm not really sure, but that does sound correct.

we might only care about the port when we're dealing with federation.

Reply:
yes, and federation will NOT matter here, the delay server is in charge of rules in its Zone only.   if it's a rule that should be run in a different Zone, it will be stored in that other Zone's catalog.

Reply:
To be honest, I've never actually tried running a remote rule on a remote zone. Could be an interesting experiment. :)

Also, zone's must all have the same port (https://docs.irods.org/4.2.9/system_overview/configuration/): 

> zone_port (required) (default 1247) - The main port used by the Zone for communication. This must be the same across all iRODS servers in a Zone.

Reply:
nice find.